content/guide/core/http.md

```diff
@@ -0,0 +1,288 @@
+      content: JSON.stringify({
+          username: "tony",
+          password: "pass"
+      })
+    }).then((response) => {
+    const result = response.content?.toJSON();
+}, (e) => {
+  // error
+})
+```
+
+## API
+
+### getString()
+
+```ts
+Http.getString(url: string): Promise<string>
+```
+Downloads the content from the specified URL as a string.
+
+---
+### getJSON()
+
+```ts
+Http.getJSON<T>(url: string): Promise<T>
+```
+
+Downloads the content from the specified URL as a string and returns its JSON.parse representation.
+
+---
+
+### getImage()
+```ts
+Http.getImage(url: string): Promise<ImageSource>
+```
+Downloads the content from the specified URL and attempts to decode it as an image.
+
+---
+
+### getFile()
+```ts
+Http.getFile(url: string, destinationFilePath?: string): Promise<File>
+```
+Downloads the content from the specified URL and attempts to save it as file. `destinationFilePath` is an optional location of where you want to save the returned file.
+
+---
+
+### request()
+```ts
+Http.request(options: HttpRequestOptions): Promise<HttpResponse>
+```
+
+Makes a generic http request using the provided options and returns a [HttpResponse](#httpresponse-interface) Object.
+
+---
+
+### HttpRequestOptions interface
+
+The HttpRequestOptions interface has the following members:
+
+#### url
+```ts
+const requestOptions = {
+  url: "https://httpbin.org"
+}
+```
+
+A string value representing the request url.
+
+---
+
+#### method
+
+```ts
+const requestOptions = {
+  url: "https://httpbin.org",
+  method: "POST"
+}
+```
+Gets or sets the request method.
+
+---
+
+#### headers
+
+```ts
+const requestOptions = {
+ 
+  headers: { "header-name" : "header-value" },
+}
+```
+_Optional_: gets or sets the request headers in JSON format.
+
+---
+
+#### content
+
+```ts
+const requestOptions = {
+ 
+  content: formData,
+}
+
+//or
+
+const requestOptions = {
+ 
+  content: "some string",
+}
+```
+_Optional_: gets or sets the request body.
+
+---
+
+#### timeout
+```ts
+const requestOptions = {
+  timeout : 22333
+}
+```
+
+_Optional_: gets or sets the request timeout in milliseconds.
+
+---
+
+#### dontFollowRedirects
+
+```ts
+const requestOptions = {
+  dontFollowRedirects : true
+}
+```
+
+_Optional_: boolean value that sets wether to **not** follow server's redirected request.
+
+---
+
+
+### HttpResponse interface
+
+#### statusCode
+
+Gets the response status code.
+
+_Type_: `number`
+
+---
+
+#### content
+
+Gets the response content.
+_Type_: [HttpContent](#httpcontent-interface) 
+
+---
+
+#### headers
+
+Gets the response headers
+
+---
+
+### HttpContent interface
+
+#### raw
+
+```ts
+response.content?.raw
+```
+
+Gets the response body as raw data.
+
+---
+
+#### toString()
+
+```ts
+response.content?.toString()
+```
+
+Gets the response body as string.
+
+---
+
+#### toJSON()
+
+```ts
+response.content?.toJSON(encoding)
+```
+
+Gets the response body as JSON object. The `encoding` is an optional of type `'UTF8' | 'GBK'`
```

Comment:
We can keep `UTF8` & `GBK` as external links and on clicking, the CTA could be more detail/some docs page for encoding types